qiskit/optimization/converters/linear_equality_to_penalty.py

```diff
@@ -35,6 +35,9 @@ def __init__(self, penalty: Optional[float] = None) -> None:
         """
```

Comment:
Given the additional types that can now be passed as penalty param I think it needs to change from
```
penalty: Optional[float] = None
```
to
```
penalty: Optional[Union[float, List[float]]] = None
```
where int is implied by virtue of having float there so does not need to be specific.